### ISSUE TITLE
StackOverflow when having more than 85 warehouses

### ISSUE THREAD

**Issue by Xzelsius** [nopSolutions/nopCommerce#1909]
We get StackOverflow exceptions when having more than 85 warehouses. If we go down to 80 warehouses the error doesn't occurs.

The exception occurs after rendering /Admin/Product/Create or /Admin/Product/Edit/{id}
They aren't thrown directly by NopCommerce, they are thrown by either System.Web or System.Runtime.Serialization or mscorlib.

Additionally, they are thrown after the client receives the response.

On the client every request (incl. ajax calls) was successfully completed with http code 200.
But the IIS itself crashes.

~~

NopCommerce 3.8
IIS Express 10 and IIS 8 / 8.5
Clean Database with no sample data.

**Comment by AndreiMaz** [comment]
please provide the full error stack

**Comment by Xzelsius** [comment]
The exception (and detail) only contains this

```
System.StackOverflowException was unhandled
Message: An unhandled exception of type 'System.StackOverflowException' occurred in System.Web.dll
```

And that is what Visual Studio shows me as call stack
[StackTrace.txt @ justbeamit.com](https://www.justbeamit.com/8eysu)


I can provide you my database if that helps.

**Comment by AndreiMaz** [comment]
@Xzelsius yes, please share your database backup

**Comment by Xzelsius** [comment]
@AndreiMaz with the most recent 3.9 version the bug does no longer exist. So i diffed our 3.8 version with the 3.9 version and found the reason for this stack overflow.

3.9 Version "Administration/Views/Product/_CreateOrUpdate.Info.cshtml" has a `</td>` at line 1517 which does not exist in the 3.8 version.

If i add the `</td>`  to our 3.8 version the stack overflow no longer occurs.
Seems like you can close this issue as you already fixed it with #2036

**Comment by AndreiMaz** [comment]
@Xzelsius thanks!
